api/net/Keycloak/CssHelper.cs

```diff
@@ -141,11 +141,9 @@ private async Task AddOrUpdateUserAsync(string username, Entities.User? user, TN
 
             // Fetch the roles for the user
             var userRoles = await _cssService.GetRolesForUserAsync(key);
+            if (userRoles.Users.Length > 1) throw new NotAuthorizedException($"Keycloak has multiple users with the same username '{key}'");
             if (user == null)
             {
-                var kUser = userRoles.Users.FirstOrDefault() ?? throw new NotAuthorizedException("The user does not exist in keycloak");
```

Comment:
If a user had no roles in Keycloak Gold realm their API will not return the user at all.